homeassistant/components/androidtv/config_flow.py

```diff
@@ -168,22 +172,22 @@ async def async_step_user(
 
     @staticmethod
     @callback
-    def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlow:
+    def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlowHandler:
         """Get the options flow for this handler."""
         return OptionsFlowHandler(config_entry)
 
 
-class OptionsFlowHandler(OptionsFlow):
+class OptionsFlowHandler(OptionsFlowWithConfigEntry):
     """Handle an option flow for Android TV."""
 
     def __init__(self, config_entry: ConfigEntry) -> None:
         """Initialize options flow."""
-        self.config_entry = config_entry
+        super().__init__(config_entry)
 
-        apps = config_entry.options.get(CONF_APPS, {})
-        det_rules = config_entry.options.get(CONF_STATE_DETECTION_RULES, {})
-        self._apps: dict[str, Any] = apps.copy()
-        self._state_det_rules: dict[str, Any] = det_rules.copy()
+        self._apps: dict[str, Any] = self.options.setdefault(CONF_APPS, {})
+        self._state_det_rules: dict[str, Any] = self.options.setdefault(
+            CONF_STATE_DETECTION_RULES, {}
+        )
```

Comment:
I think that the dict `copy` here is still required, otherwise the option flow will write directly the option dict referenced during the flow before saving.

Reply:
This is now handled in the base class, where a deep-copy is created during initialisation:
https://github.com/home-assistant/core/blob/cb06f8b66882037e81ffee94afd7ccdb3969e338/homeassistant/config_entries.py#L1696-L1702